### DISCUSSION TITLE
courses/beginner/10

### DISCUSSION THREAD

**Discussion by giscus** [mitre/saf-training#258] [Ideas]
# courses/beginner/10

Profile Dependencies (Overlays) In addition to its own controls, an InSpec profile can leverage controls from one or more other InSpec profiles. When a profile depends on contro...

https://mitre-saf-training.netlify.app/courses/beginner/10.html

**Comment by p-oneil** [comment]
Inputs file discussion? How it works in an overlay. Best practices.

**Comment by jrodriguez617** [comment]
18. Clearly explain the rationale and use cases for overlays and dependencies. While the section describes how overlays work and how to implement them, it would benefit from a more explicit explanation of why overlays and dependencies are valuable. Discuss scenarios where overlays solve real problems, such as customizing controls for different environments, maintaining compliance across multiple systems, or streamlining updates from upstream profiles. This helps users understand when and why to use overlays, not just how.

**Comment by jrodriguez617** [comment]
19. More coverage of control modification and inheritance risks. The content briefly touches on modifying controls (e.g., changing impact) but does not discuss the potential risks or limitations. Expand on how modifications can affect compliance, reporting, and auditability. Warn users about the risk of unintentionally overriding important logic or metadata, and provide guidance on tracking changes and maintaining documentation when controls are modified in overlays.

**Comment by jrodriguez617** [comment]
20. Address version management and upstream profile updates. The section mentions that overlays keep profiles in sync, but does not explain how versioning works or what happens when upstream profiles change. Add content about best practices for managing profile versions, handling breaking changes, and testing overlays after upstream updates. Encourage users to specify versions or tags in dependencies, and to regularly review and update overlays to ensure continued compatibility and compliance.
